collator/src/lib.rs

```diff
@@ -256,18 +258,24 @@ impl<P, E> Worker for CollationNode<P, E> where
 		let client = service.client();
 		let network = service.network();
 		let known_oracle = client.clone();
+		let select_chain = if let Some(select_chain) = service.select_chain() {
+			select_chain
+		} else {
+			info!("The node cannot work because it can't select chain.");
+			return Box::new(future::ok(()));
```

Comment:
hmm better to return an empty future (which will never end) no ? or an unwrap

Reply:
I'd suggest doing the [`or_or_else` as in the PR](https://github.com/paritytech/substrate/pull/2574/files#diff-d2b444bb5d594e411359291f8be4fe56R512)

Reply:
changed to return an error ()